spec/components/attachment_link_spec.rb

```diff
@@ -62,4 +62,28 @@ def component_name
     )
     assert_select "a.govuk-link[data-gtm='attachment-preview']"
   end
+
+  it "does not have a newline character after the last metadata attribute's closing span tag" do
+    render_component(
+      attachment: {
+        title: "Attachment",
+        url: "attachment",
+        content_type: "text/plain",
+      },
+    )
+
+    expect(rendered).to end_with("</span>)</span>")
+  end
+
+  it "only has one newline character after the link element" do
+    render_component(
+      attachment: {
+        title: "Attachment",
+        url: "attachment",
+      },
+    )
+
+    expect(rendered).not_to match(/<\/a>\n\n/)
+    expect(rendered).to match(/<\/a>\n/)
```

Comment:
I feel like we actually want this to also not a have new line as a I believe that can also generate a space. Ugh inline element markup can be super annoying!

Reply:
In Publishing API the links do still have a newline, although I don't really understand why it's needed!

Reply:
Now no longer adding any newlines after the link element 👍